src/resources/filters/normalize/pandoc3.lua

```diff
@@ -2,38 +2,95 @@
 -- Copyright (C) 2020-2023 Posit Software, PBC
 
 function parse_pandoc3_figures() 
-  return {
-    Figure = function(fig)
-      if (#fig.content == 1 and fig.content[1].t == "Plain") then
-        
-        if #fig.content[1].content == 1 and fig.content[1].content[1].t == "Image" then
-          -- "pandoc 2 normalization"
-          local image = fig.content[1].content[1]
-          image.classes:extend(fig.classes)
-          for k, v in pairs(fig.attributes) do
-            image.attributes[k] = v
-          end
-          if fig.identifier ~= "" then
-            image.identifier = fig.identifier
-          end
+  local walk_recurse
+  walk_recurse = function(constructor)
+    local drop_figure_treatment = function(el)
+      return el:walk(walk_recurse(pandoc.Plain))
+    end
+    return {
+      traverse = "topdown",
+      BulletList = drop_figure_treatment,
+      BlockQuote = drop_figure_treatment,
+      Table = drop_figure_treatment,
+      OrderedList = drop_figure_treatment,
+      Note = drop_figure_treatment,
+      Figure = function(fig)
+        if (#fig.content == 1 and fig.content[1].t == "Plain") then
           
-          return pandoc.Para(image)
+          if #fig.content[1].content == 1 and fig.content[1].content[1].t == "Image" then
+            -- "pandoc 2 normalization"
+            local image = fig.content[1].content[1]
+            image.classes:extend(fig.classes)
+            for k, v in pairs(fig.attributes) do
+              image.attributes[k] = v
+            end
+            if fig.identifier ~= "" then
+              image.identifier = fig.identifier
+            end
+            
+            return constructor(image)
+          else
+            -- if user filters muck with images, we need to support this as well.
+            -- however, we can't forward figure information along the image, so
+            -- this is necessarily a best-effort situation until we truly fix figures.
+            return constructor(fig.content[1].content)
+          end
         else
-          -- if user filters muck with images, we need to support this as well.
-          -- however, we can't forward figure information along the image, so
-          -- this is necessarily a best-effort situation until we truly fix figures.
-          return pandoc.Para(fig.content[1].content)
+          print("Couldn't parse figure:")
```

Comment:
Tiny thing- there is a Lua `error` function which is intended for printing errors like this. Not sure if it is in scope here...

Reply:
This is good, thanks. I made the change.